cns/service/main.go

```diff
@@ -773,18 +779,78 @@ func InitializeMultiTenantController(ctx context.Context, httpRestService cns.HT
 	return nil
 }
 
-// initializeCRD state
-func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cnsconfig configuration.CNSConfig) error {
-	var requestController singletenantcontroller.RequestController
+type nodeNetworkConfigGetter interface {
+	Get(context.Context) (*v1alpha.NodeNetworkConfig, error)
+}
 
-	logger.Printf("[Azure CNS] Starting request controller")
+type ncStateReconciler interface {
+	ReconcileNCState(ncRequest *cns.CreateNetworkContainerRequest, podInfoByIP map[string]cns.PodInfo, scalar v1alpha.Scaler, spec v1alpha.NodeNetworkConfigSpec) cnstypes.ResponseCode
+}
 
-	kubeConfig, err := kubecontroller.GetKubeConfig()
+// TODO(rbtr) where should this live??
+// InitCNS initializes cns by passing pods and a createnetworkcontainerrequest
+func initCNS(ctx context.Context, cli nodeNetworkConfigGetter, ncReconciler ncStateReconciler) error {
+	// Get nnc using direct client
+	nnc, err := cli.Get(ctx)
 	if err != nil {
-		logger.Errorf("[Azure CNS] Failed to get kubeconfig for request controller: %v", err)
+
+		if crd.IsNotDefined(err) {
+			return errors.Wrap(err, "failed to get NNC during init CNS state")
+		}
+
+		if nnc == nil {
+			logger.Errorf("NodeNetworkConfig is not present on cluster")
+			return nil
```

Comment:
is this just a sanity check or is it expected to happen? seems like the client code would not return nil if I'm reading correctly

Reply:
you're right, this will never be nil with the new client and the subsequent `IgnoreNotFound(err)` check covers this case anyway